deeplabcut/refine_training_dataset/stitch.py

```diff
@@ -723,40 +723,41 @@ def stitch(self, add_back_residuals=True):
                 for node in self.G
                 if node not in ("source", "sink")
             )
-            if (
-                incomplete_tracks == 1
-            ):  # All remaining nodes must belong to the same track
-                # Verify whether there are overlapping tracklets
-                for t1, t2 in combinations(remaining_nodes, 2):
-                    if t1 in t2:
-                        # Pick the segment that minimizes "smoothness", computed here
-                        # with the coefficient of variation of the differences.
-                        if t1 in remaining_nodes:
-                            remaining_nodes.remove(t1)
-                        if t2 in remaining_nodes:
-                            remaining_nodes.remove(t2)
-                        track = sum(remaining_nodes)
-                        hyp1 = track + t1
-                        hyp2 = track + t2
-                        dx1 = np.diff(hyp1.centroid, axis=0)
-                        cv1 = dx1.std() / np.abs(dx1).mean()
-                        dx2 = np.diff(hyp2.centroid, axis=0)
-                        cv2 = dx2.std() / np.abs(dx2).mean()
-                        if cv1 < cv2:
-                            remaining_nodes.add(t1)
-                            self.residuals.append(t2)
-                        else:
-                            remaining_nodes.add(t2)
-                            self.residuals.append(t1)
-                paths.append(list(remaining_nodes))
-            elif incomplete_tracks > 1:
-                # Rebuild a full graph from the remaining nodes without
-                # temporal constraint on what tracklets can be stitched together.
-                self.build_graph(list(remaining_nodes), max_gap=np.inf)
-                self.G.nodes["source"]["demand"] = -incomplete_tracks
-                self.G.nodes["sink"]["demand"] = incomplete_tracks
-                _, self.flow = nx.capacity_scaling(self.G)
-                paths += self.reconstruct_paths()
+            if len(remaining_nodes) > 0:
```

Comment:
Great! By the way, I checked, and we could also have:
```
if not remaining_nodes:
    return
```
as the `finally` clause is executed before the function exits, no matter what.
But that's a detail 😃